### PR TITLE
Improves the behavior of the `create` command

### PR DESCRIPTION
**Pull request by HavokInspiration**
Currently, the `create` commands behaves a bit differently than bake does.

This PR syncs them to a similar state :
- filename created by the `create` command are now CamelBack as file generated by bake (I had to copy/paste the entire execute method of the original create command to change the position of the creation class `postMigrationCreation` "callback" that renames the _snake_case filename that phinx generates).
- the template the `create` command outputs is now more in sync with what bake outputs : it extends the good abstract class. Basically, if you do a `bin/cake bake migration TestCreate` and a `bin/cake migrations create TestCreate`, you now get the same file.

### CODE REVIEW

**Comment by HavokInspiration** [comment]
Refs #143, #264, #265.

**Comment by dereuromark** [comment]
> had to copy/paste the entire execute method of the original create command

Oh geeze - that method on shinx side was definitely too long and doing to much.
Terrible...
And we have to pay the price now.

Can we maybe at least remove some of the things like yoda bad practice etc?
Or are you afraid syncing with the parent method content will be more difficult from here on?

**Comment by HavokInspiration** [comment]
I may have thought of another way to implement this without having to copy/paste the entire method (and avoid having to sync with the original one).
I'll give this a test tomorrow or during the week-end.

**Comment by HavokInspiration** [comment]
Strike that last one, that might difficult as I won't have the filepath generated by phinx.
I'd rather not edit the method too much (I fixed the basic errors Travis picked up), keeping it the way it is will make the process of syncing easier as you mentionned.

**Comment by HavokInspiration** [comment]
Actually, there might be a way. Since classname have to be unique, when using create, if the file has been create, it's likely to be the only one with that name. So I could `glob()` a file with the name argument in the migrations path and found it to rename it. I would the just call `parentExecute` as with other command and do the renaming stuff after that.
I'll give this a try.

**Comment by HavokInspiration** [comment]
I tried a different implementation to have a consistency between the name from bake and the name from the migrations `create` command that prevents us from having to copy / paste the entire phinx method just to move some event. This should be better to sync with phinx in case the base method implementation changes.

**Comment by josegonzalez** [comment]
Can we force delete the old one? I do this sometimes when I screw up creating the migration.

File: src/ConfigurationTrait.php [hunk 1]
```diff
@@ -83,13 +83,14 @@ public function getConfig($forceRefresh = false)
         $connectionConfig = ConnectionManager::config($connection);
         $adapterName = $this->getAdapterName($connectionConfig['driver']);
 
+        $templatePath = Plugin::path('Migrations') . 'src' . DS . 'Template';
```
Comment (dereuromark) [inline src/ConfigurationTrait.php]: Can we have the DS attached here at the end? Paths in CakePHP usually always end this way.
Further down then

    'file' => $templatePath . 'Phinx' . DS . 'create.php.template'

**Comment by HavokInspiration** [comment]
@josegonzalez I'm not a fan of force-deleting the old one. If it's just a mistake (like you have a lot of migrations), you might end up deleting an old one.
But we can have an interactive shell asking what to do and a parameter (I think) to instruct bake to delete the old one without asking permission.

**Comment by HavokInspiration** [comment]
Hm... Wait... @josegonzalez is your question related to this PR or the old one ? I'm not sure I understand it if in the context of this one.

**Comment by josegonzalez** [comment]
Oh, it was meant for the other one, my bad.

**Comment by HavokInspiration** [comment]
Ok my answer was with the other PR in mind.

**Comment by HavokInspiration** [comment]
I took care of your comment @dereuromark

File: tests/comparisons/Create/TestCreate.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+use Migrations\AbstractMigration;
+
+class TestCreate extends AbstractMigration
```
Comment (HavokInspiration) [inline tests/comparisons/Create/TestCreate.php]: I'll just ignore this comment, the `comparisons` folder will be ignored in the stickler file from #301

**Comment by lorenzo** [comment]
Great, the difference between both commands was annoying for me :)

**Pull request merged**